app.py

```diff
@@ -1,7 +1,141 @@
-from flask import Flask, render_template
-app = Flask(__name__)
+"""
+A Flask application for snapcraft.io.
+
+The web frontend for the snap store.
+"""
+
+import flask
+import requests
+import requests_cache
+import datetime
+import json
+import humanize
+from dateutil import parser
+
+
+app = flask.Flask(__name__)
+uncached_session = requests.Session()
+cached_session = requests_cache.CachedSession(expire_after=60)
+request_timeout = 2
+snap_details_url = (
+    "https://api.snapcraft.io/api/v1/snaps/details/{snap_name}"
+    "?channel=stable"
+)
 
 
 @app.errorhandler(404)
-def page_not_found(e):
-    return render_template('404.html'), 404
+def page_not_found(error):
+    """
+    For 404 pages, display the 404.html template,
+    passing through the error description.
+    """
+
+    return flask.render_template(
+        '404.html', description=error.description
+    ), 404
+
+
+@app.route('/<snap_name>/')
+def snap_details(snap_name):
+    """
+    A view to display the snap details page for specific snaps.
+
+    This queries the snapcraft API (api.snapcraft.io) and passes
+    some of the data through to the snap-details.html template,
+    with appropriate sanitation.
+    """
+
+    query_headers = {
+        'X-Ubuntu-Series': '16',
```

Comment:
Snap revisions are architecture-specific (the vast majority), so it the API. The `X-Ubuntu-Architecture: <arch>` defaults to 'amd64' and is required for arch-specific fields like `channel_maps_list`.
It will certainly have UI implications.

Reply:
Ah, so by default we're only going to see snaps that exist for `amd64`? In that case I'd rather provide the `X-Ubuntu-Architecture` header, again to make it clear. I'll do that. Thanks.